### ISSUE TITLE
[Feature Request]: Support JDBC Type 4 Driver for MO

### ISSUE THREAD

**Issue by domingozhang** [matrixorigin/matrixone#2386]
### Is there an existing issue for the same feature request?

- [X] I have checked the existing issues.

### Describe the feature you'd like

Type 4 JDCBC driver is widely used to MySQL in both development and operations. Discussed with internal and external developers, this is crucial for their work and will also strengthen MO's ability.

Details are as following:
https://docs.oracle.com/cd/E13157_01/wlevs/docs30/jdbc_drivers/usedriver.html

**Comment by daviszhen** [comment]
It has a close relationship with the catalog. The catalog should provide system variables, meta data tables,etc. It may also need features like SSL, role,user,privilige in 0.5.

**Comment by florashi181** [comment]
Agreed. This issue can start when the catalog is ready, and there is a high probability that it will not be completed in 0.5 release.

**Comment by florashi181** [comment]
@domingozhang confirm the status of this feature request?  I see davis have done sth, but not sure it is all done. Let's decide this issue move to v0.6 or not.
